c7n/resources/cw.py

```diff
@@ -315,3 +316,74 @@ def process_resource_set(self, client, accounts, resources):
             if found:
                 results.append(r)
         return results
+
+
+@LogGroup.action_registry.register('set-encryption')
+class EncryptLogGroup(BaseAction):
+    """Encrypt/Decrypt a log group
+
+    :example:
+
+    .. code-block: yaml
+
+        policies:
+          - name: encrypt-log-group
+            resource: log-group
+            filters:
+              - kmsKeyId: absent
+            actions:
+              - type: set-encryption
+                kms_key: kms:key:arn
+                state: True
+
+          - name: decrypt-log-group
+            resource: log-group
+            filters:
+              - kmsKeyId: kms:key:arn
+            actions:
+              - type: set-encryption
+                state: False
+    """
+    schema = type_schema(
+        'set-encryption',
+        kms_key={'type': 'string'},
```

Comment:
minor: `kms_key` is redundant here, I prefer just `key` as the only `key` that can be specified here is kms and you're already referencing key as `self.key` when processing resources anyway

Reply:
yeah, I can change that. I thought it was in line with other actions that used kms as a parameter, but maybe I was wrong.